vignettes/satterthwaite.Rmd

```diff
@@ -119,4 +119,68 @@ of freedom is calculated for each of them. Finally, using properties of the F
 distribution's expectation, the denominator degree of freedom for the whole F statistic
 is derived.
 
+## Satterthwaite degrees of freedom for sandwich covariance
+
+In @bell2002bias the degrees of freedom of sandwich covariance are described.
+
+### One-dimensional contrast
+
+For one-dimensional contrast, following the same notation in [Details of the model fitting in `mmrm`](algorithm.html)
+and [Details of the Kenward-Roger calculations](kenward.html), we have the following derivation.
+For an estimator of variance with the following term
+
+\[
+  v = s c(X^\top X)^{-1}\sum_{i}{X_i^\top A_i \epsilon_i \epsilon_i^\top A_i X_i} (X^\top X)^{-1} c^\top
+\]
+
+where $s$ takes the value of $\frac{n}{n-1}$, $1$ or $\frac{n-1}{n}$, and $A_i$ takes $I_i$, $(I_i - H_{ii})^{-\frac{1}{2}}$, or $(I_i - H_{ii})^{-1}$
+respectively, $c$ is a vector, then $v$ can be decomposed into the a weighted sum of independent $\Chi_1^2$ distribution, where the weights are
+the eigen values of $n\times n$ matrix $G$, its element can be represented as
+\[
+  G_{ij} = g_i V g_j
+\]
+
+where
+
+\[
+  g_i = s^{\frac{1}{2}} (I - H)_i^\top A_i X_i (X^\top X)^{-1} c
+\]
+\[
+  H = X(X^\top X)^{-1}X^\top
+\]
+
+$(I - H)_i$ corresponds to the rows of subject $i$.
+
+So the degrees of freedom can be represented as
+\[
+  \nu = \frac{(\sum_{i}\lambda_i)^2}{\sum_{i}{\lambda_i^2}}
+\]
+
+where $\lambda_i$ is the eigen value of matrix $G$.
+@bell2002bias also suggests that $V$ can be chosen as identify matrix, so $G_{ij} = g_i  g_j$.
+
+Following [Weighted Least Square Estimator](algorithm.html#weighted-least-squares-estimator), we can transform
+the original $X$ into $\tilde{x}$ to use the above equations.
+
+To avoid repeated computation of matrix $A_i$, $H$ etc for different contrasts, we calculate and cache the following
+
+\[
+  G^\ast_i = (I - H)_i^\top A_i X_i (X^\top X)^{-1}
+\]
+is a $\sum_i{m_i} \times p$ matrix. With different contrasts, we need only calculate the following
+\[
+  g_i = G^\ast_i c
+\]
+to obtain a $\sum_i{m_i} \times 1$ matrix, $G$ can be computed with $g_i$.
+
+To obtain the degrees of freedom, and to avoid eigen computation on a large matrix, we can use the following equation
```

Comment:
is this from the paper? would be good to cite/derive

Reply:
derivation added